tests/unittests/test_retry_logic.py

```diff
@@ -42,6 +42,34 @@ def test_retries_on_500(self):
         # 5 is the max tries specified in the tap
         self.assertEquals(5, mocked_account.get_ad_creatives.call_count )
 
+    def test_retries_on_503(self):
+        """`AdCreative.sync.do_request()` calls a `facebook_business` method,
+        `get_ad_creatives()`, to make a request to the API. We mock this
+        method to raise a `FacebookRequestError` with an `http_status` of
+        `503`.
+
+        We expect the tap to retry this request up to 5 times, which is
+        the current hard coded `max_tries` value.
+        """
+
+        # Create the mock and force the function to throw an error
+        mocked_account = Mock()
+        mocked_account.get_ad_creatives = Mock()
+        mocked_account.get_ad_creatives.side_effect = FacebookRequestError(
+            message='',
+            request_context={"":Mock()},
+            http_status=503,
+            http_headers=Mock(),
+            body={}
+        )
```

Comment:
To mock actual exception, can we assign `string` value to the body?

```suggestion
        mocked_account.get_ad_creatives.side_effect = FacebookRequestError(
            message='',
            request_context={"":Mock()},
            http_status=503,
            http_headers=Mock(),
            body="Service Unavailable"
        )
```

Reply:
done